driver-core/src/main/com/mongodb/internal/operation/BulkWriteBatch.java

```diff
@@ -282,21 +281,11 @@ private BulkWriteResult getBulkWriteResult(final BsonDocument result) {
     }
 
     private List<BulkWriteInsert> getInsertedItems(final BsonDocument result) {
-        if (payload.getPayloadType() == SplittablePayload.Type.INSERT) {
```

Comment:
I removed the conditional just to simplify the code.  If the payload type is anything but INSERT, then `payload.getInsertedIds` will be empty anyway.